ksqldb-cli/src/main/java/io/confluent/ksql/cli/console/UnclosedQuoteChecker.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.cli.console;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Checks to see if the line is in the middle of a quote.  Unlike
+ * org.jline.reader.impl.DefaultParser, this is comment aware.
+ */
+public final class UnclosedQuoteChecker {
+  private static final String COMMENT = "--";
+
+  private UnclosedQuoteChecker() {}
+
+  public static boolean isUnclosedQuote(final String line) {
+    requireNonNull(line, "line");
```

Comment:
fyi: no need to null check an argument you're not storing in your object state.

And I think if you remove this you should also remove the `line != null` in the `for` loop.